src/AggregateFunctions/AggregateFunctionMap.cpp

```diff
@@ -0,0 +1,133 @@
+#include "AggregateFunctionMap.h"
+#include "AggregateFunctions/AggregateFunctionCombinatorFactory.h"
+#include "Functions/FunctionHelpers.h"
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+}
+
+class AggregateFunctionCombinatorMap final : public IAggregateFunctionCombinator
+{
+public:
+    String getName() const override { return "Map"; }
+
+    DataTypes transformArguments(const DataTypes & arguments) const override
```

Comment:
I think we should throw an exception when we have map as first argument and arguments.size() > 1

Reply:
And it would be better to make all arguments validation here to assume that we have valid arguments in `transformAggregateFunction`. See coments in IAggregateFunctionCombinator:
https://github.com/ClickHouse/ClickHouse/blob/ea97eee326a94c9a23d1b4349f660a17ad29b551/src/AggregateFunctions/IAggregateFunctionCombinator.h#L37-L41
https://github.com/ClickHouse/ClickHouse/blob/ea97eee326a94c9a23d1b4349f660a17ad29b551/src/AggregateFunctions/IAggregateFunctionCombinator.h#L55-L60

Reply:
Maybe it would be better to make all arguments validation here and remove it from `transformAggregateFunction`? Then we can assume that all arguments are valid in `transformAggregateFunction`:
https://github.com/ClickHouse/ClickHouse/blob/ea97eee326a94c9a23d1b4349f660a17ad29b551/src/AggregateFunctions/IAggregateFunctionCombinator.h#L58-L60

Reply:
Very good points, done.